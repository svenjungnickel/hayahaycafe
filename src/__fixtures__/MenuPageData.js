export default {
    slug: 'slug',
    title: 'title',
    subtitle: 'subtitle',
    headerImage: {
        publicURL: 'publicURL',
    },
    images: [
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'Menu page 01',
            title: 'Menu page 01',
        },
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'Menu page 02',
            title: 'Menu page 02',
        },
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'Menu page 03',
            title: 'Menu page 03',
        },
    ],
    meta: {
        title: 'meta title',
        description: 'meta description.',
        keywords: 'meta keywords',
    },
};
