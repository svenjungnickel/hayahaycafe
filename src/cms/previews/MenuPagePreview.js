import React from 'react';
import PropTypes from 'prop-types';
import { MenuPageTemplate } from '../../templates/MenuPage';

const MenuPagePreview = ({ entry, widgetFor }) => (
    <MenuPageTemplate
        title={entry.toJS().data.title}
        subtitle={entry.toJS().data.subtitle}
        headerImage={entry.toJS().data.headerImage}
        content={widgetFor('body')}
        images={entry.toJS().data.images}
    />
);

MenuPagePreview.propTypes = {
    entry: PropTypes.shape({
        toJS: PropTypes.func,
    }).isRequired,
    widgetFor: PropTypes.func.isRequired,
};

export default MenuPagePreview;
