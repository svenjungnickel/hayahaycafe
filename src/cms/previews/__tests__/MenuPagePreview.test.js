import React from 'react';
import renderer from 'react-test-renderer';
import MenuPagePreview from '../MenuPagePreview';
import MenuPageData from '../../../__fixtures__/MenuPageData';

jest.mock('../../../components/Header');
jest.mock('../../../components/Separator');
jest.mock('../../../components/Image');

describe('MenuPagePreview', () => {
    it('renders without entry throws prop type error', () => {
        const renderMenuPagePreview = () => {
            const widgetFor = () => {};

            const component = <MenuPagePreview widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderMenuPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders without widgetFor throws prop type error', () => {
        const renderMenuPagePreview = () => {
            const entry = {
                toJS: () => {},
            };

            const component = <MenuPagePreview entry={entry} />;
            renderer.create(component);
        };

        expect(renderMenuPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid entry throws prop type error', () => {
        const renderMenuPagePreview = () => {
            const entry = {
                toJS: 123,
            };
            const widgetFor = () => {};

            const component = <MenuPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderMenuPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid widgetFor throws prop type error', () => {
        const renderMenuPagePreview = () => {
            const entry = {
                toJS: () => {},
            };
            const widgetFor = {};

            const component = <MenuPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderMenuPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const entry = {
            toJS: () => {
                return {
                    data: MenuPageData,
                };
            },
        };
        const widgetFor = () => 'content';

        const component = <MenuPagePreview entry={entry} widgetFor={widgetFor} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
