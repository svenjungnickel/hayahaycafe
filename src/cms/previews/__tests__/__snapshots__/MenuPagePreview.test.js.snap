// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuPagePreview renders correctly 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="menu container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
          data-cy="menuContent"
        >
          <div>
            content
          </div>
          <hr />
        </div>
        <div
          className="menu__image col-lg-6 col-md-12"
          data-cy="menuImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
        <div
          className="menu__image col-lg-6 col-md-12"
          data-cy="menuImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
